src/backend/src/services/change-request.services.ts

```diff
@@ -119,17 +123,24 @@ export default class ChangeRequestsService {
         const updatedDuration = foundCR.wbsElement.workPackage.duration + foundPs.timelineImpact;
 
         const changes = [
-          {
-            changeRequestId: crId,
-            implementerId: reviewer.userId,
-            detail: buildChangeDetail('Budget', String(wpProj.budget), String(newBudget))
-          },
-          {
-            changeRequestId: crId,
-            implementerId: reviewer.userId,
-            detail: buildChangeDetail('Duration', String(foundCR.wbsElement.workPackage.duration), String(updatedDuration))
-          }
+          createChangeJsonNonList(
+            'Budget',
+            String(wpProj.budget),
+            String(newBudget),
+            crId,
+            reviewer.userId,
+            foundCR.wbsElement.wbsElementId
+          ),
+          createChangeJsonNonList(
+            'Duration',
+            String(foundCR.wbsElement.workPackage.duration),
+            String(updatedDuration),
+            crId,
+            reviewer.userId,
+            foundCR.wbsElement.wbsElementId
+          )
         ];
+        changes.filter((a: any): boolean => a !== undefined);
```

Comment:
I think this line should be either expanded or is unnecessary. Underneath this we are accessing changes[1] and changes[0] so filtering it out might cause an index-out-of-bounds type error.

Reply:
Try writing a test where the request does not change the budget/timeline, see what happens with this code (does it blow up)